### ISSUE TITLE
Auditor mode v2

### ISSUE THREAD

**Issue by guillaume-byte** [GrayboxTech/weightslab#99]
Right now, the audit freezes the model, but still does the slow backward operation in the train loop. 
We should wrap the backward somehow and skip it if we are in the audit mode, i.e., audit should be faster in this way.

**Comment by guillaume-byte** [comment]
For partial backward, i.e., training from a subset of data:
```import torch
import torch.nn.functional as F

logits = model(x)  # [B, C]
per_sample = F.cross_entropy(logits, y, reduction="none")  # [B]

mask = torch.ones_like(per_sample)
mask[2] = 0.0  # ignore sample 2

loss = (per_sample * mask).sum() / mask.sum().clamp_min(1.0)
loss.backward()```

**Comment by guillaume-byte** [comment]
Right now, there is no logger distinction between audit mode and train mode; this diff. comes from the UI, i.e., received logs during audit? If restarted experiment from the previous, we lose this information.
